src/openmmio.py

```diff
@@ -888,7 +919,27 @@ def update_simulation(self, **kwargs):
         else:
             self.create_simulation(**self.simkwargs)
 
-    def set_positions(self, shot=0, traj=None):
+    def set_restraint_positions(self, shot):
+        """
+        Set reference positions for energy restraints.  This may be a different set of positions
+        from the "current" positions that are stored in self.mol and self.xyz_omm.
+        """
+        if hasattr(self, 'restrain_k'):
+            ## Generate OpenMM-compatible positions in nanometers
+            xyz = self.ref_mol.xyzs[shot] / 10.0
+            have_restraint_frc = False
+            for frc in self.simulation.system.getForces():
+                if isinstance(frc, CustomExternalForce):
```

Comment:
I think this may have a potential risk to mix up with a OpenMM system that has an CustomExternalForce defined for other purpose. I think a safer and faster way to access the force is to save the index when the force is added to the system. For example in line 528 above:
```
self.restraint_frc_index = self.system.addForce(restraint_frc)
```
then here we can do
```
frc = self.simulation.system.getForce(self.restraint_frc_index)
```

Reply:
Changes have been made in the latest commit.